ruth/simulator/singlenode.py

```diff
@@ -145,6 +147,19 @@ def compute_alternatives(self, alternatives_providers: List[AlternativesProvider
         combined_alts = remove_infinity_alternatives(combined_alts, self.sim.routing_map)
         return combined_vehicles, combined_alts
 
+    def select_routes(self, route_selection_providers: List[RouteSelectionProvider], vehicles: List[VehicleWithPlans]):
```

Comment:
No need for this to be a method, it could just be a function. Makes it clearer that it doesn't depend on the simulator and doesn't use `self`.